docs/platforms/python/profiling/index.mdx

```diff
@@ -8,109 +8,121 @@ sidebar_order: 5000
     dsn="___PUBLIC_DSN___",
-
-    # Add data like request headers and IP for users, if applicable;
-    # see https://docs.sentry.io/platforms/python/data-management/data-collected/ for more info
     send_default_pii=True,
-
     traces_sample_rate=1.0,
-
-    # To set a uniform sample rate
-    # Set profiles_sample_rate to 1.0 to profile 100%
-    # of sampled transactions.
-    # We recommend adjusting this value in production,
-    profiles_sample_rate=1.0,
-
-    # Alternatively, to control sampling dynamically
-    profiles_sampler=profiles_sampler
+    # To collect profiles for all profile sessions,
+    # set `profile_session_sample_rate` to 1.0.
+    profile_session_sample_rate=1.0,
+    # Profiles will be automatically collected while
+    # there is an active span.
+    profile_lifecycle="trace",
 )
 ```
 
-<Alert>
-
-The <PlatformIdentifier name="profiles_sample_rate" /> setting is _relative_ to the <PlatformIdentifier name="traces_sample_rate" /> setting.
+### Enabling Manual Lifecycle Profiling
 
-For Profiling to work, you have to first enable [Sentry’s tracing](/concepts/key-terms/tracing/) via `traces_sample_rate` (like in the example above). Read our <PlatformLink to="/tracing/">tracing setup documentation</PlatformLink> to learn how to configure sampling. If you set your sample rate to 1.0, all transactions will be captured.
+```python
+import sentry_sdk
 
-</Alert>
+sentry_sdk.init(
+    dsn="___PUBLIC_DSN___",
+    send_default_pii=True,
+    traces_sample_rate=1.0,
 
-### Upgrading from Older Python SDK Versions
+    # To collect profiles for all profile sessions,
+    # set `profile_session_sample_rate` to 1.0.
+    profile_session_sample_rate=1.0,
+    # Profiles will be collected when
+    # `sentry_sdk.profiler.start_profiler` is called and
+    # stopped when `sentry_sdk.profiler.stop_profiler` is called.
+    profile_lifecycle="manual",
+)
 
-Profiling was experimental in SDK versions `1.17.0` and older. Learn how to upgrade <PlatformLink to="/profiling/troubleshooting/#ipgrading-from-older-sdk-versions">here</PlatformLink>.
+sentry_sdk.profiler.start_profiler()
 
-## Enable Continuous Profiling
+# run some code here
 
-<Include name="feature-stage-beta.mdx" />
+sentry_sdk.profiler.stop_profiler()
+```
 
-_(New in version 2.24.1)_
+### Managing profile sampling rates
 
-The current profiling implementation stops the profiler automatically after 30 seconds (unless you manually stop it earlier). Naturally, this limitation makes it difficult to get full coverage of your app's execution. We now offer an experimental continuous mode, where profiling data is periodically uploaded while running, with no limit on how long the profiler may run.
+Sentry SDK supports an additional `profile_session_sample_rate` that will enable or disable profiling for the entire session. This can be used if you want to control session sampling rates at the service level as the sampling decision is evaluated only once at SDK init.
```

Comment:
This wording makes it sound like the parameter is optional, but it has to be set to >0 for profiling to work at all